src/core/extra-file-descriptor.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include "list.h"
+#include "macro.h"
+
+typedef struct ExtraFileDescriptor {
+        int fd;
+        char *fdname;
```

Comment:
So, hmm, this is a reduced version of `ServiceFDStore`. I wonder if we should just make this generic, and wrap FDStore around this.

Reply:
True, I guess I was a little nervous to do that because ServiceFDStore is FDs stored across service restarts for all unit types and ExtraFileDescriptors are FDs not stored across service restarts and only used for StartTransientUnit. So I was a bit wary to overload the meaning of FDStore.

We could make another fdstore instance like `ServiceFDStore *extra_fd_store;` and ignore event_source and do_poll...I will try and see how simple/complex the code looks. Thanks for the suggestion!

Reply:
Tried both approaches and I felt both made the first approach made the code harder to read. The 2nd approach didn't really work because you have to end up including the LIST_FIELDS() field in ServiceFD, so you either add all fields to ServiceExtraFD (equivalent to first approach) or add a layer of indirection to ServiceExtraFD.

Of course, I'm a new (attempted) contributor, so feel free to overrule me :)